buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -140,14 +139,6 @@ public static void configureCompile(Project project) {
         java.setSourceCompatibility(BuildParams.getMinimumRuntimeVersion());
         java.setTargetCompatibility(BuildParams.getMinimumRuntimeVersion());
 
-        Function<File, String> canonicalPath = file -> {
-            try {
-                return file.getCanonicalPath();
-            } catch (IOException e) {
-                throw new GradleException("Failed to get canonical path for " + file, e);
-            }
-        };
-
         project.afterEvaluate(p -> {
```

Comment:
TODO: the only left reason for this afterEvaluate is that compileOptions.get/setArguments(...) is not using the provider api. Discuss this and maybe raise a ticket at gradle core.

Reply:
no concrete plans from the gradle core team except "Before the heat death of the universe, yes."